assignments/array-methods.js

```diff
@@ -54,30 +57,53 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names 
+// into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(element) {
+    fullName.push(`${element.first_name} ${element.last_name}`);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. 
+// Convert each first name into all caps and log the result
+let allCaps = runners.map(elem => elem.first_name.toUpperCase());
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they 
+// can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of 
+// L and log the result
+let largeShirts = runners.filter(elem => elem.shirt_size === "L" );
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the 
+// result
+let ticketPriceTotal = runners.reduce( (acc, curr) => acc + curr.donation. 0 );
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve 
+// given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Find the companies who donated over $200 to feature these sponsors:
+let goldSponsors = runners.filter(elem => elem.donation >= 200);
```

Comment:
Should be `elem.donation > 200` since it is asking for companies who donated *over* $200.